cmake/utils.cmake

```diff
@@ -5,3 +5,18 @@ macro(ternary var boolean value1 value2)
         set(${var} ${value2})
     endif()
 endmacro()
+
+# Function to extract #define values from C/C++ headers
+function(extract_define_from_header header_file define_name output_var)
+    file(STRINGS "${header_file}" define_line REGEX "#define ${define_name}")
+
+    # Handle string defines: #define FOO "value"
+    if(define_line MATCHES "\"([^\"]*)\"")
```

Comment:
this might incorrectly match something like `#define FOO 123 // this is some "bar" stuff` no?